### ISSUE TITLE
Peer client hangs when server responds to upgrade request with non 101

### ISSUE THREAD

**Issue by AdamMagaluk** [zettajs/zetta#158]
While testing in front of a proxy i've noticed that when the proxy responds with an error code to the upgrade request zetta peer_client hangs instead of trying to reconnect.

**Comment by kevinswiber** [comment]
+1 I noticed this, as well.

Sent from my iPhone

On Feb 28, 2015, at 9:45 AM, Adam Magaluk <notifications@github.com> wrote:

While testing in front of a proxy i've noticed that when the proxy responds
with an error code to the upgrade request zetta peer_client hangs instead
of trying to reconnect.

—
Reply to this email directly or view it on GitHub
<https://github.com/zettajs/zetta/issues/158>.

**Comment by AdamMagaluk** [comment]
I thinks it's an easy fix by listening for 'response' event on the request.

```js
req.on('response', function(res) {
  self.emit('close', res.statusCode);
});
```

**Comment by AdamMagaluk** [comment]
Fixed in v0.22.0
